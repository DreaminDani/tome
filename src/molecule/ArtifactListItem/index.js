import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles, Paper, Typography } from '@material-ui/core';
import Link from 'next/link';
import { getDisplayNameFromName } from '../../helpers';

const useStyles = makeStyles(theme => ({
  root: {
    listStyle: 'none',
    '&:hover': {
      cursor: 'pointer',
    },
  },
  paper: {
    padding: theme.spacing(3, 2),
    marginBottom: theme.spacing(1),
  },
}));

function ArtifactListItem(props) {
  const classes = useStyles();

  // eslint-disable-next-line react/destructuring-assignment
  const { id, name, auth_metadata, created_at } = props.artifact;

  const createdDate = new Date(created_at);

  return (
    <li id={id} className={classes.root}>
      <Link href="/artifact/[slug]" as={`/artifact/${id}`}>
        <Paper className={classes.paper}>
          <Typography variant="h5" component="h3">
            {name}
          </Typography>
          <Typography component="p">
            <strong>Created by: </strong>
            {getDisplayNameFromName(auth_metadata.name)} on{' '}
            {createdDate.toLocaleDateString()} at{' '}
            {createdDate.toLocaleTimeString()}
          </Typography>
        </Paper>
      </Link>
    </li>
  );
}

ArtifactListItem.propTypes = {
  artifact: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    user_id: PropTypes.number,
    created_at: PropTypes.string,
    updated_at: PropTypes.string,
    auth_metadata: PropTypes.object,
  }),
};

export default ArtifactListItem;
