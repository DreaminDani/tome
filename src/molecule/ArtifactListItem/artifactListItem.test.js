/* eslint-env jest */

import { shallow } from 'enzyme';
import ArtifactListItem from '.';

describe('ArtifactListItem', () => {
  const fakeArtifact = {
    id: 'abc',
    name: 'Test',
    user_id: 1,
    created_at: '2015-03-25T12:00:00Z',
    updated_at: '2015-03-25T12:00:00Z',
    auth_metadata: {},
  };

  let artifactListItem;
  beforeEach(() => {
    artifactListItem = shallow(<ArtifactListItem artifact={fakeArtifact} />);
  });

  it("should display the artifact's name", () => {
    expect(artifactListItem.find({ component: 'h3' }).prop('children')).toEqual(
      fakeArtifact.name
    );
  });

  it('should display details about the artifact', () => {
    expect(
      artifactListItem.find({ component: 'p' }).prop('children')
    ).toMatchSnapshot();
  });
});
