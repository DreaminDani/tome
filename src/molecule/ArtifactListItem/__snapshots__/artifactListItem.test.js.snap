// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtifactListItem should display details about the artifact 1`] = `
Array [
  <strong>
    Created by: 
  </strong>,
  "A tome user",
  " on",
  " ",
  "3/25/2015",
  " at",
  " ",
  "12:00:00 PM",
]
`;
